boards/posix/native_posix/hw_models_top.c

```diff
@@ -92,12 +98,17 @@ static void hwm_sleep_until_next_timer(void)
 		/* LCOV_EXCL_STOP */
 	}
 
-	if (signaled_end || (simu_time > end_of_time)) {
+	_SAFE_CALL(pthread_mutex_lock(&exit_mutex));
+	if (signaled_end || (simu_time > end_of_time) || hwm_exit_request) {
+		int exit_code = hwm_exit_request ? hwm_exit_code : 0;
```

Comment:
Minor: `hwm_exit_code` is 0 unless it was set differently in `hwm_main_loop_exit` (with `hwm_exit_request`). So you may as well have `int exit_code = hwm_exit_code;` (without the if), and add a comment instead if you feel its needed.